### PR TITLE
Fix flaky streaming test

### PR DESCRIPTION
**Pull request by JamesNK**
Fix failure noticed here - https://github.com/grpc/grpc-dotnet/pull/2131#issuecomment-1551555902

### CODE REVIEW

**Comment by JamesNK** [comment]
Please review 🙏

File: test/Grpc.Net.Client.Tests/Balancer/ConnectionManagerTests.cs [hunk 1]
```diff
@@ -115,12 +116,14 @@ public async Task PickAsync_ChannelStateChangesWithWaitForReady_WaitsForCorrectE
...
 
+        await BalancerWaitHelpers.WaitForSubchannelsToBeReadyAsync(logger, clientChannel, 0);
+
         var pickTask2 = clientChannel.PickAsync(
             new PickContext { Request = new HttpRequestMessage() },
             waitForReady: true,
             CancellationToken.None).AsTask().DefaultTimeout();
 
-        Assert.IsFalse(pickTask2.IsCompleted);
+        Assert.IsFalse(pickTask2.IsCompleted, "PickAsync should wait until an subchannel is ready.");
```
Comment (BrennanConroy) [inline test/Grpc.Net.Client.Tests/Balancer/ConnectionManagerTests.cs:126]: I don't get it. You wait for subchannels to be ready, assert that PickAsync doesn't complete, and then say PickAsync shouldn't complete until subchannels are ready. (Also, the grammar is wrong in the sentence, I assume you mean "any subchannels are ready" or "a subchannel is ready")

Comment (BrennanConroy) [reply test/Grpc.Net.Client.Tests/Balancer/ConnectionManagerTests.cs:126]: Ah wait, now I understand, the `expectedCount` value is `0`, maybe use named params.

**Review by BrennanConroy — APPROVED** [review]

File: test/Grpc.Net.Client.Tests/Balancer/ConnectionManagerTests.cs [hunk 2]
```diff
@@ -115,12 +116,14 @@ public async Task PickAsync_ChannelStateChangesWithWaitForReady_WaitsForCorrectE
             transportFactory.Transports[i].UpdateState(ConnectivityState.TransientFailure);
         }
 
+        await BalancerWaitHelpers.WaitForSubchannelsToBeReadyAsync(logger, clientChannel, 0);
```
Comment (JamesNK) [inline test/Grpc.Net.Client.Tests/Balancer/ConnectionManagerTests.cs:119]:
```suggestion
        await BalancerWaitHelpers.WaitForSubchannelsToBeReadyAsync(logger, clientChannel, expectedCount: 0);
```

**Pull request merged**